### PR TITLE
[TT-3361] Deprecating the enable_health_checks option

### PR DESCRIPTION
**Pull request by fletcherjy**
The enable_health_checks is deprecated so this PR is to update the documentation.

### CODE REVIEW

File: tyk-docs/content/planning-for-production/planning-for-production.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ As well as changing the default secrets - see [Change all the shared secrets](/d
 
 ### Health checks are expensive
 
-In order to keep real-time health-check data and make it available to the Health-check API, Tyk needs to record information for every request, in a rolling window - this is an expensive operation and can limit throughput - you have two options: switch it off, or get a box with more cores.
+In order to keep real-time health-check data and make it available to the Health-check API, Tyk needs to record information for every request, in a rolling window - this is an expensive operation and can limit throughput.  Therefore [enable_health_checks](/docs/tyk-oss-gateway/configuration/) option is deprecated from version 3.0.7 and 3.2.2 and disabled by default from version 4.0.  It is recommended to either remove the configuration or set it to false for earlier versions.
```
Comment (buger) [inline tyk-docs/content/planning-for-production/planning-for-production.md:97]: I would put WAY more emphasis on it. 

Literally: Ensure that this option disabled.

File: tyk-docs/content/shared/gateway-config.md [hunk 2]
```diff
@@ -848,19 +848,19 @@ Type: `bool`<br />
...
 ### health_check
-This section enables the configuration of the health-check API endpoint and the size of the sample data cache (in seconds).
+Deprecated. This section enables the configuration of the health-check API endpoint and the size of the sample data cache (in seconds).
 
 ### health_check.enable_health_checks
 EV: **TYK_GW_HEALTHCHECK_ENABLEHEALTHCHECKS**<br />
 Type: `bool`<br />
 
-Setting this value to `true` will enable the health-check endpoint on /Tyk/health.
+Deprecated. Setting this value to `true` will enable the health-check endpoint on /Tyk/health.
```
Comment (buger) [inline tyk-docs/content/shared/gateway-config.md:848]: Lets wrap it to **bold**

**Review by buger — APPROVED** [review]

File: tyk-docs/content/shared/gateway-config.md (hunk 2 above)
Comment (marksou) [reply tyk-docs/content/shared/gateway-config.md:848]: @buger Should this be done in the Tyk repo version now? I'll raise a PR there.

Comment (marksou) [reply tyk-docs/content/shared/gateway-config.md:848]: @buger https://github.com/TykTechnologies/tyk/pull/4038

File: tyk-docs/content/planning-for-production/planning-for-production.md (hunk 1 above)
Comment (zalbiraw) [reply tyk-docs/content/planning-for-production/planning-for-production.md:97]: I am confused. We discuss why we need it and we also discuss that its expensive. Then we say we are deprecating it? What happened to the health checks?

**Review by zalbiraw — CHANGES_REQUESTED** [review]

**Comment by letzya** [comment]
@zalbiraw it is not deprecated in the code...
Here's why https://github.com/TykTechnologies/tyk/pull/4038/files
@buger can you please advise

**Pull request closed without merging**